src/java.base/share/classes/sun/security/rsa/RSASignature.java

```diff
@@ -205,27 +208,22 @@ protected boolean engineVerify(byte[] sigBytes) throws SignatureException {
         }
         try {
             if (sigBytes.length != RSACore.getByteLength(publicKey)) {
-                throw new SignatureException("Signature length not correct: got " +
+                throw new SignatureException("Bad signature length: got " +
                     sigBytes.length + " but was expecting " +
                     RSACore.getByteLength(publicKey));
             }
             byte[] digest = getDigestValue();
             byte[] decrypted = RSACore.rsa(sigBytes, publicKey);
-            byte[] unpadded = padding.unpad(decrypted);
+            RSAPadding.Output po = padding.unpad(decrypted);
```

Comment:
In case you are already here, what if comparing the padded/encoded result, without use unpad() any longer. I meant to follow the spec as described in RFC8017#section-8.2.2: encode the `decryped` bytes and then compare the result with the `digest` bytes.

Reply:
Ok, I can switch to this approach.